src/main/scala/scorex/transaction/TransactionFactory.scala

```diff
@@ -88,21 +90,33 @@ object TransactionFactory {
     tx <- ReleaseSlotsTransaction.create(senderPrivateKey, request.slotId, request.fee, request.feeScale, time.getTimestamp())
   } yield tx
     
-  def createContract(request: CreateContractRequest, wallet: Wallet, time: Time): Either[ValidationError, CreateContractTransaction] = for {
+  def registerContract(request: RegisterContractRequest, wallet: Wallet, time: Time): Either[ValidationError, RegisterContractTransaction] = for {
     senderPrivateKey <- wallet.findPrivateKey(request.sender)
-    contract <- Contract.buildContract(request.content, request.name, true)
-    tx <- CreateContractTransaction.create(senderPrivateKey, contract, request.fee, request.feeScale, time.getTimestamp())
+    contract <- Contract.fromBase58String(request.contract)
+    initData <- DataEntry.fromBase58String(request.initData)
+    tx <- RegisterContractTransaction
+      .create(senderPrivateKey,
+        contract,
+        initData,
+        request.description.filter(_.nonEmpty).getOrElse(Deser.deserilizeString(Array.emptyByteArray)),
+        request.fee,
+        request.feeScale,
+        time.getTimestamp())
   } yield tx
 
-  def changeContractStatus(request: ChangeContractStatusRequest, action: ChangeContractStatusAction.Value, wallet: Wallet, time: Time): Either[ValidationError, ChangeContractStatusTransaction] = for {
+  def executeContractFunction(request: ExecuteContractFunctionRequest, wallet: Wallet, time: Time): Either[ValidationError, ExecuteContractFunctionTransaction] = for {
     senderPrivateKey <- wallet.findPrivateKey(request.sender)
-    tx <- ChangeContractStatusTransaction.create(senderPrivateKey, request.contractName, action, request.fee, request.feeScale, time.getTimestamp())
-  } yield tx
-
-  def broadcastChangeContractStatus(request: SignedChangeContractStatusRequest, action: ChangeContractStatusAction.Value): Either[ValidationError, ChangeContractStatusTransaction] = for {
-    _signature <- ByteStr.decodeBase58(request.signature).toOption.toRight(ValidationError.InvalidRequestSignature)
-    _sender <- PublicKeyAccount.fromBase58String(request.senderPublicKey)
-    tx <- ChangeContractStatusTransaction.create(_sender, request.contractName, action, request.fee, request.feeScale, request.timestamp, _signature)
+    contractId <- ContractAccount.fromString(request.contractId)
+    functionData <- DataEntry.fromBase58String(request.functionData)
+    tx <- ExecuteContractFunctionTransaction
+      .create(senderPrivateKey,
+        contractId,
+        request.functionIndex,
+        functionData,
+        request.attachment.filter(_.nonEmpty).map(Base58.decode(_).get).getOrElse(Array.emptyByteArray),
```

Comment:
can Base58.decode() return an error?

Reply:
Base58.decode => Try[Array[Byte]]

this code follow the origin payment and transfer function format.

Reply:
okay